tests/core/engine_adapter/integration/test_integration_clickhouse.py

```diff
@@ -0,0 +1,522 @@
+import typing as t
+import pytest
+from tests.core.engine_adapter.integration import TestContext
+import pandas as pd
+from sqlglot import exp, parse_one
+
+
+@pytest.fixture(
+    params=[
+        pytest.param(
+            "clickhouse",
+            marks=[
+                pytest.mark.docker,
+                pytest.mark.engine,
+                pytest.mark.clickhouse,
+                pytest.mark.xdist_group("engine_integration_clickhouse"),
+            ],
+        ),
+        pytest.param(
+            "clickhouse_cluster",
+            marks=[
+                pytest.mark.docker,
+                pytest.mark.engine,
+                pytest.mark.clickhouse_cluster,
+                pytest.mark.xdist_group("engine_integration_clickhouse_cluster"),
+            ],
+        ),
+        pytest.param(
+            "clickhouse_cloud",
+            marks=[
+                pytest.mark.engine,
+                pytest.mark.remote,
+                pytest.mark.clickhouse_cloud,
+                pytest.mark.xdist_group("engine_integration_clickhouse_cloud"),
+            ],
+        ),
+    ]
+)
+def mark_gateway(request) -> t.Tuple[str, str]:
+    return request.param, f"inttest_{request.param}"
+
+
+@pytest.fixture
+def test_type() -> str:
+    return "query"
+
+
+def _get_source_queries_and_columns_to_types(
+    ctx,
+    insert_table: exp.Table,
+    target_table: exp.Table,
+    columns_to_types: t.Dict[str, exp.DataType] = {
+        "id": exp.DataType.build("Int8", dialect="clickhouse"),
+        "ds": exp.DataType.build("Date", dialect="clickhouse"),
+    },
+):
+    return ctx.engine_adapter._get_source_queries_and_columns_to_types(
+        parse_one(f"SELECT * FROM {insert_table.sql()}"),  # type: ignore
+        columns_to_types,
+        target_table=target_table.sql(),
+    )
+
+
+def _create_table_and_insert_existing_data(
+    ctx: TestContext,
+    existing_data: pd.DataFrame = pd.DataFrame(
+        [
+            {"id": 1, "ds": "2024-01-01"},
+            {"id": 2, "ds": "2024-02-01"},
+            {"id": 3, "ds": "2024-02-28"},
+            {"id": 4, "ds": "2024-03-01"},
+        ]
+    ),
+    columns_to_types: t.Dict[str, exp.DataType] = {
+        "id": exp.DataType.build("Int8", "clickhouse"),
+        "ds": exp.DataType.build("Date", "clickhouse"),
+    },
+    table_name: str = "data_existing",
+    partitioned_by: t.Optional[t.List[exp.Expression]] = [
+        parse_one("toMonth(ds)", dialect="clickhouse")
+    ],
+) -> exp.Table:
+    existing_data = existing_data
+    existing_table_name: exp.Table = ctx.table(table_name)
+    ctx.engine_adapter.ctas(
+        existing_table_name.sql(),
+        ctx.input_data(existing_data, columns_to_types),
+        columns_to_types,
+        partitioned_by=partitioned_by,
+    )
+    return existing_table_name
+
+
+def test_insert_overwrite_by_condition_replace_partitioned(ctx: TestContext):
+    ctx.init()
```

Comment:
`ctx.init()` can also be removed, it happens automatically in the fixture function now